reify.go

```diff
@@ -305,6 +315,7 @@ func reifyGetField(
 	opts fieldOptions,
 	name string,
 	to reflect.Value,
+	baseType reflect.Type,
```

Comment:
we use `baseType` to indicate that we have already chased down the type. Maybe name it `fieldType` to indicate that this is the actual type of the field in the struct.